Sources/TuistLoader/Utils/PluginsTemplatePathHelper.swift

```diff
@@ -0,0 +1,55 @@
+import Foundation
+import TSCBasic
+import TuistCore
+import TuistGraph
+import TuistSupport
+
+protocol PluginsTemplatePathHelping {
+    func templatePath(
+        for pluginName: String,
+        resourceName: String,
+        resourceSynthesizerPlugins: [ResourceSynthesizerPlugin]
+    ) throws -> AbsolutePath
+}
+
+enum PluginsHelperError: FatalError, Equatable {
+    case pluginNotFound(String, [String])
+    case resourceTemplateNotFound(name: String, plugin: String)
+
+    var type: ErrorType {
+        switch self {
+        case .pluginNotFound,
+             .resourceTemplateNotFound:
+            return .abort
+        }
+    }
+
+    var description: String {
+        switch self {
+        case let .pluginNotFound(name, availablePlugins):
+            return "Plugin \(name) was not found. Available plugins: \(availablePlugins.joined(separator: ", "))"
+        case let .resourceTemplateNotFound(name: name, plugin: pluginName):
+            return "No template \(name) found in a plugin \(pluginName)"
+        }
+    }
+}
+
+final class PluginsTemplatePathHelper: PluginsTemplatePathHelping {
+    func templatePath(
+        for pluginName: String,
+        resourceName: String,
+        resourceSynthesizerPlugins: [ResourceSynthesizerPlugin]
```

Comment:
The name `ResourceSynthesizerPlugin` is a bit misleading. Does it represent a resource synthesizer contained in a plugin? In that case I think `PluginResourceSynthesizer` would be more accurate.

Reply:
I am not sure about this one, I think we should stick to the same naming `{nameOfPlugin}Plugin` - e.g. `ProjectDescriptionHelpersPlugin` has also `Plugin` as suffix, not prefix. Or do you think that one should be changed as well?

Reply:
Maybe it's me, but I think it's more descriptive. This is how I read both:
- `xxxPlugin`: A plugin of type `xxx`.
- `Pluginxxxx`: A `xxx` component contained in a plugin.

Considering Tuist are boxes that encapsulate Tuist components, I think the latter is more accurate. No need to change it in this PR, but I think we should rename it before we continue adding more elements to plugins. This will be misleading down the road because contributors will think that helpers and resource synthesizers are plugins, and not that they are contained in plugin, which is the idea we'd like to convey.